File: saksflyt-api/src/test/java/no/nav/melosys/saksflytapi/domain/MelosysEessiMeldingTest.java
package no.nav.melosys.saksflytapi.domain;

import no.nav.melosys.domain.eessi.melding.MelosysEessiMelding;
import org.assertj.core.api.Assertions;
import org.jeasy.random.EasyRandom;
import org.jeasy.random.EasyRandomParameters;
import org.junit.jupiter.api.Test;

public class MelosysEessiMeldingTest {
    @Test
    public void testSerialisering() {
        EasyRandomParameters easyRandomParameters = new EasyRandomParameters().collectionSizeRange(1, 2).stringLengthRange(1, 4);
        MelosysEessiMelding eessiMelding = new EasyRandom(easyRandomParameters).nextObject(MelosysEessiMelding.class);
        Prosessinstans p = new Prosessinstans();
        p.setData(ProsessDataKey.EESSI_MELDING, eessiMelding);
        final MelosysEessiMelding deserialisering = p.getData(ProsessDataKey.EESSI_MELDING, MelosysEessiMelding.class);
        Assertions.assertThat(deserialisering).isEqualToComparingFieldByField(eessiMelding);
    }
}

File: saksflyt-api/src/test/kotlin/no/nav/melosys/saksflytapi/domain/MelosysEessiMeldingTest.kt
package no.nav.melosys.saksflytapi.domain

import io.kotest.matchers.shouldBe
import no.nav.melosys.domain.eessi.melding.MelosysEessiMelding
import org.jeasy.random.EasyRandom
import org.jeasy.random.EasyRandomParameters
import org.junit.jupiter.api.Test

class MelosysEessiMeldingTest {
    @Test
    fun `serialisering og deserialisering av EESSI melding skal bevare data`() {
        val easyRandomParameters = EasyRandomParameters().collectionSizeRange(1, 2).stringLengthRange(1, 4)
        val eessiMelding = EasyRandom(easyRandomParameters).nextObject(MelosysEessiMelding::class.java)
        val p = Prosessinstans.forTest { }

        p.setData(ProsessDataKey.EESSI_MELDING, eessiMelding)
        val deserialisering = p.getData(ProsessDataKey.EESSI_MELDING, MelosysEessiMelding::class.java)

        deserialisering shouldBe eessiMelding
    }
}
